src/codegen/optimizer.lisp

```diff
@@ -206,6 +209,8 @@
    node
    canonicalize
    (match-dynamic-extent-lift env)
+   ;; FIXME: Is this the right place to propagate-constants?
+   (propagate-constants env)
```

Comment:
I had put it here intuiting that it should happen as early as possible so other passes can benefit from it. At the same time, it should not change the correctness for any of the passes. I did not put it before the `match-dynamic-extent-lift` just to be on the safe side.

Reply:
@eliaslfox any thoughts?

Reply:
The PR looks good to me as it is, but I agree that it would be good for @eliaslfox to take a look and make sure it is in the right place relative to the rest of the optimizations (and also verify that updating the `node-rator-type` is helpful here for `node-direct-application` and maybe also `node-application`?)